### PR TITLE
Simplified storage index: removed config hash id in favor of using a unique model identifier

### PR DESCRIPTION
**Pull request by anton164**
This should simplify the use of the storage. All scripts and data was updated according.

Also updated the `store_summary_metrics` to behave as PUT request to ensure that other metrics are not overwritten.

## BEFORE CHANGE Example snippet for reading summaries and updating metrics
```
from sumtool.storage import get_summaries

generated_summary_lookup = {
  "BERTS2S": "2221cf6c3bad26c64a12417282dabf3c",
  "TranS2S": "adaab4ed6990eeae0ea8e98c78a722fe",
  "PtGen": "5641d358ff7a45cc65592938f1020675", 
  "TConvS2S": "dfadf4c1e1847488969892cc58be06c1"
}

# dictionary of doc id -> summary data
gold_summaries = get_summaries("xsum", "Gold")["dfd16d395e88b38829ff2d5b98743404"]

for model_name, model_id in data_lookup.items():
  # dictionary of doc id -> summary data
  model_generated_summaries = get_summaries("xsum", model_name)[model_id]

  # compute metrics, ...i.e. compare generated summaries with gold summaries 
  summary_metrics = compute_metrics(...)

  store_summary_metrics("xsum", model_name, model_id, summary_metrics)
```

## AFTER CHANGE Example snippet for reading summaries and updating metrics
```
from sumtool.storage import get_summaries

# dictionary of doc id -> summary data
gold_summaries = get_summaries("xsum", "maynez-gold")

for model_id in ["maynez-berts2s", "maynez-ptgen", ...]:
  # dictionary of doc id -> summary data
  model_generated_summaries = get_summaries("xsum", model_id)

  # compute metrics, ...i.e. compare generated summaries with gold summaries 
  summary_metrics = compute_metrics(...)

  store_summary_metrics("xsum", model_name, model_id, summary_metrics)
```

### CODE REVIEW

File: README.md
Comment (dleve123) [inline README.md:75]:
```suggestion
#### `/data/<dataset>/<model-id>-summaries.json`
```

File: data/xsum/data.txt
Comment (dleve123) [inline data/xsum/data.txt:1]: I think this was just scratch work, should this be deleted?

File: interface/model_interface.py
Comment (dleve123) [inline interface/model_interface.py:8]: why the empty `st.write()`?

File: interface/model_interface.py
Comment (dleve123) [inline interface/model_interface.py:15]: good catch 👍

File: sumtool/storage.py
Comment (dleve123) [inline sumtool/storage.py:32]:
```suggestion
        the slugified string
```

File: sumtool/storage.py
Comment (dleve123) [inline sumtool/storage.py:18]: minor: seems like your adding types to the other functions, want to add them here?

File: sumtool/storage.py
Comment (dleve123) [inline sumtool/storage.py:78]: why this change?

File: data/xsum/facebook-bart-large-xsum-config.json
Comment (dleve123) [inline data/xsum/facebook-bart-large-xsum-config.json:85]: so this `sumtool_id` is effectively the "foreign key" for the name of the metric, summary JSON files, correct? It _should_ correspond to the name of this config file, but it really could be anything? Is that correct?

**Review by dleve123 — CHANGES_REQUESTED** [review]
overall looks good! caught a couple of typos that we should fix before merging and left a question about some of the DB structures.

File: data/xsum/data.txt
Comment (anton164) [reply data/xsum/data.txt:1]: Yes, good catch

File: sumtool/storage.py
Comment (anton164) [reply sumtool/storage.py:78]: Because we no longer index summaries by the `model_config_hash`. They are indexed directly by the summary id.

**Comment by anton164** [comment]
Thanks for the review @dleve123 ! Your comments have been addressed

**Review by celine-lee — APPROVED** [review]

**Review by dleve123 — APPROVED** [review]

**Comment by anton164** [comment]
Blocked by #78 - will merge after that one to resolve conflicts (lmk when you merge it @dleve123)

**Comment by dleve123** [comment]
@anton164 #78 should be good to go once someone approves the changes I made in response to PR feedback.

**Comment by dleve123** [comment]
@anton164 I don't think we're seeing any git merge conflicts here, but I believe there's a semantic conflict to be addressed: removing/updating the main key in the metrics JSON documents ([example](https://github.com/factula/sumtool/blob/main/data/xsum/BERTS2S-metrics.json#L2)). Is that correct?

**Comment by anton164** [comment]
@dleve123 correct, I needed to migrate the entailment script & data to the updated API. Done now

**Pull request merged**
